plugins/plugin-java/che-plugin-java-ext-jdt/org-eclipse-core-resources/src/main/java/org/eclipse/core/resources/ResourcesPlugin.java

```diff
@@ -189,12 +190,12 @@
   @Inject
   public ResourcesPlugin(
       @Named("che.jdt.workspace.index.dir") String indexPath,
-      @Named("che.user.workspaces.storage") String workspacePath,
+      RootDirPathProvider pathProvider,
```

Comment:
what if keep in constructor
```@Named("che.workspaces.storage") String workspacePath```
and will use provider for this property
```
    bind(String.class)
        .annotatedWith(Names.named("che.workspaces.root.dir"))
        .toProvider(RootDirPathProvider.class);
```

Reply:
why to make this logic more complex?

what needed is to inject configuration object (called RootDirPathProvider)?

Reply:
If you do not mind, I'll put in my five cents. Though technically there is almost no difference between two cases, I find it more comfortable to remember, to navigate and to use (in IDE) `RootDirPathProvider` class instead of a variable named  "che.workspaces.root.dir" or "che.workspace.storage", it's much easier to access provider class (or property) documentation and so on. Probably it is caused by the way object oriented programmer thinks - in objects and classes, not properties and values.

Reply:
ок